File: leetcode/#136.js
// question --> https://leetcode.com/problems/single-number/

/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    let ans = 0;
    for (let index = 0; index < nums.length; index++) {
        ans ^= nums[index];
    }
    return ans
};

// TC: O(N); SC: O(1)
// end

File: leetcode/#136.py
# question --> https://leetcode.com/problems/single-number/

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ans = 0
        for num in nums:
            ans ^= num
        return ans
        
        # TC: O(N); SC: O(1) 
        
# end